### PR TITLE
Complete task 5

### PR DESCRIPTION
**Pull request by evTyurin**

### CODE REVIEW

File: src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java [hunk 1]
```diff
@@ -36,19 +35,28 @@ public void addCraftsman(Craftsman craftsman) {
...
     @Override
-    public void removeCraftsmen(Craftsman craftsman) {
-        craftsmanDAO
-                .getCraftsmen()
-                .remove(craftsman);
+    public void removeCraftsmanById(long id) {
+
+        List<Craftsman> craftsmen = craftsmanDAO.getCraftsmen();
+
+        for (Craftsman craftsman : craftsmen) {
```
Comment (mrseneca) [inline src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java]: Use stream API

File: src/com/senlainc/warsaw/tyurin/service/GaragePlaceService.java [hunk 2]
```diff
@@ -36,8 +37,18 @@ public void addGaragePlace(GaragePlace garagePlace) {
     }
 
     @Override
-    public void deleteGaragePlace(GaragePlace garagePlace) {
-        garagePlaceDAO.deleteGaragePlace(garagePlace);
+    public void removeGaragePlace(long id) {
+
+        List<GaragePlace> garagePlaces = garagePlaceDAO.getGaragePlaces();
+
+        for (GaragePlace garagePlace : garagePlaces) {
```
Comment (mrseneca) [inline src/com/senlainc/warsaw/tyurin/service/GaragePlaceService.java]: Same - stream API

**Review by mrseneca — CHANGES_REQUESTED** [review]
Small fixes with streams needed

File: src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java (hunk 1 above)
Comment (evTyurin) [reply src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java]: The idea was that I break the loop when found craftsman I need.
Some unneeded iterations could happen if I use stream API

Comment (mrseneca) [reply src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java]: The stream API has vertical execution of operations (except sorting). This means that once the element passes the filter, find first will return a value and end the stream. Additional operations are not required

https://akifhatipoglu.medium.com/guide-to-java-stream-api-stream-execution-order-4c2df14d1d00

**Review by mrseneca — APPROVED** [review]
Good job. 
Made comment under for <-> Stream API use, added some article

File: src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java (hunk 1 above)
Comment (evTyurin) [reply src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java]: Now it is clear to me, thanks!

**Pull request merged**